test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -0,0 +1,339 @@
+
+	_, err = user2DynamicVWClient.Cluster(logicalcluster.Wildcard).Resource(schema.GroupVersionResource{Version: "v1", Resource: "sheriffs", Group: "wild.wild.west"}).List(ctx, metav1.ListOptions{})
+	require.ErrorContains(
+		t, err,
+		`sheriffs.wild.wild.west is forbidden: User "user-2" cannot list resource "sheriffs" in API group "wild.wild.west" at the cluster scope: virtual apiexport access not permitted`,
+		"user-2 must not be allowed to list sheriff resources")
+
+	_, err = user2DynamicVWClient.Cluster(logicalcluster.Wildcard).Resource(schema.GroupVersionResource{Version: "v1", Resource: "configmaps"}).List(ctx, metav1.ListOptions{})
+	require.NoError(t, err, "user-2 must be allowed to list native types")
+
+	t.Logf("grant access to sherrifs for %q in workspace %q", apisv1alpha1.MaximalPermissionPolicyRBACUserGroupPrefix+"user-2", serviceProvider1Workspace)
+	cr, crb = createClusterRoleAndBindings(
+		"apiexport-claimed",
+		"apis.kcp.dev:binding:user-2", "User",
+		[]string{"create", "list"},
+		"wild.wild.west", "sheriffs", "",
+	)
+	_, err = kubeClient.RbacV1().ClusterRoles().Create(logicalcluster.WithCluster(ctx, serviceProvider1Workspace), cr, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = kubeClient.RbacV1().ClusterRoleBindings().Create(logicalcluster.WithCluster(ctx, serviceProvider1Workspace), crb, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("verify that we can lists all claimed resources using a wildcard request")
+	claimedGVRs := []schema.GroupVersionResource{
+		{Version: "v1", Resource: "configmaps"},
+		{Version: "v1", Resource: "sheriffs", Group: "wild.wild.west"},
+	}
+	framework.Eventually(t, func() (success bool, reason string) {
+		for _, gvr := range claimedGVRs {
+			_, err := user2DynamicVWClient.Cluster(logicalcluster.Wildcard).Resource(gvr).List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return false, fmt.Sprintf("error while waiting to list %q: %v", gvr, err)
+			}
+		}
+		return true, ""
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "listing claimed resources failed")
+
+	t.Logf("verify that we can lists all claimed resources using a request towards the tenant workspace")
+	framework.Eventually(t, func() (success bool, reason string) {
+		_, err = user2DynamicVWClient.Cluster(tenantWorkspace).Resource(schema.GroupVersionResource{Version: "v1", Resource: "sheriffs", Group: "wild.wild.west"}).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("error while waiting to list sherriffs: %v", err)
+		}
+		return true, ""
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "listing claimed resources failed")
+
+	_, err = user2DynamicVWClient.Cluster(tenantShadowCRDWorkspace).Resource(schema.GroupVersionResource{Version: "v1alpha1", Resource: "cowboys", Group: "wildwest.dev"}).List(ctx, metav1.ListOptions{})
+	require.Error(t, err, "expected error, got none")
+	require.True(t, errors.IsNotFound(err))
+}
```

Comment:
bravo on this, holy moly is that a complex test

Reply:
uff, yes, this e2e test took most of the coding time. but it literally "tells a story".